### ISSUE TITLE
Screen sharing in iOS

### ISSUE THREAD

**Issue by mohib-sheth** [dkornev/TwilioVideoXamarinIOS#6]
I am trying to create Xamarin iOS sample for screen sharing using Twilio. I have taken reference from twilio ReplayKitExample using broadcast extension (https://github.com/twilio/video-quickstart-ios/tree/master/ReplayKitExample). 
When I create a local screen share track from custom video source (reference in the example's SampleHandler), it gives an exception:

Unable to marshal the parameter 

1 whose managed type is 'Twilio.Video.iOS.TVIVideoSink' to managed.
Additional information:
	Selector: setSink:
	Method: TwilioXamarinSample.iOS.ReplayKiVSource:set_Sink (Twilio.Video.iOS.TVIVideoSink)
Inner exception message: Cannot create an instance of Twilio.Video.iOS.TVIVideoSink because it is an abstract class.

I tried by adding the latest nuget package as well as building the binding library and adding the dll. I am working on VS Community for Mac 8.4.4. I have also attached the solution.

I further tried editing the ApiDefinition (which probably is not ideal) and removed the [abstract] definitions from the members of TVIVideoSink protocol. Doing this, I was able to create the local video track and video source and connect to the room as well. However, after this, when I convert CMSampleBuffers (received from ReplayKit) to TVIVideoFrame and pass it to sink's OnVideoFrame, it gives: 
An Exception of type 'Foundation.You_Should_Not_Call_base_In_This_Method' was thrown.

Can you help out with how a custom video source and local video track using that can be created?

[TwilioXamarinSample 2.zip](https://github.com/dkornev/TwilioVideoXamarinIOS/files/4634740/TwilioXamarinSample.2.zip)

**Comment by jtorvald** [comment]
I started on this as well but I ran into the problem that the broadcast extension didn't work. I don't see my app when I start casting. Do you have that working? I will have another look at this in a couple of weeks. Did you make any progress?

**Comment by jtorvald** [comment]
@mohib-sheth @dkornev any update on this? 
@mohib-sheth I tried your sample app and there I can see a broadcast extension so I probably did something wrong with the extensions. Anyway: I get the same error about the abstract class with my VideoSource :

Unable to marshal the parameter #1 whose managed type is 'Twilio.Video.iOS.TVIVideoSink' to managed.
Additional information:
	Selector: setSink:
	Method: TwilioXamarinSample.iOS.ReplayKitVideoSource:set_Sink (Twilio.Video.iOS.TVIVideoSink)

**Comment by mohibsheth** [comment]
@jtorvald No, unfortunately, wasn't able to resolve this either. Stuck on the same problem. Tried everything I could think of but it didn't work.
